### ISSUE TITLE
Repeated questions

### ISSUE THREAD

**Issue by dimaguy** [Nucleo-Estudantes-Informatica-ISEP/antirecurso#63]
During quizzes, it's easy to find a colision(two questions being the same) between randomly selected questions from the question bank:
![image](https://github.com/Nucleo-Estudantes-Informatica-ISEP/antirecurso/assets/7207103/d74ac90a-7d44-4534-ae90-7d3f46cc8204)
![image](https://github.com/Nucleo-Estudantes-Informatica-ISEP/antirecurso/assets/7207103/f55efc83-5a7c-4d46-b627-6af0f8e2ec2c)

Even if the answers are ordered differently, this probably isn't expected behavior.

**Comment by tomasflopes** [comment]
The backend effectively prevents duplicated questions from appearing in the same quiz. In the scenario you are describing (and in similar cases), the [question crawler](https://github.com/tomasflopes/questions-crawler) retrieved the same question from different exams and stored it in the database. To address this, our proposed solution is to eliminate questions with identical content, including both the question and options. It's worth noting that this may not completely resolve the issue, as some variations (string matching issues) may still exist.

While we acknowledge this issue, we currently do not plan on fixing it in the near future. The reason is that addressing this problem would involve removing existing exam answers from old exams. Perhaps, in the future, as we undergo a database reset, we could consider addressing this issue.


 We appreciate your feedback and encourage further suggestions or insights!

**Comment by dimaguy** [comment]
Instead of string matching issues, this case could probably benefit of [string similarity algorithms](https://yassineelkhal.medium.com/the-complete-guide-to-string-similarity-algorithms-1290ad07c6b7) to make some choices, this could be done as a cleanup task in the db after adding entries or on the fly mid quiz creation. Mind you that asking the same question in different ways is not inherently bad, as it tests if the student is choosing at random or actually knows what they are doing, but the same question with a high similarity coefficient is not as useful in the quiz

And instead of removing existing stuff from old exams, I propose soft-deletions, an extra column in the database that marks whether a question is eligible to be shown to the student or not (and only one instance of the question would show that), that way you'll be able to keep historical accuracy while providing a functional quiz

**Comment by tomasflopes** [comment]
Yes, using string similarity algorithms could potentially address the issue, but it wouldn't entirely eliminate the need for manual verification of each match. We are cautious about introducing a feature that might inadvertently remove valuable questions in the process. This is the primary reason for the prolonged delay in implementing this feature, as we are currently prioritizing improvements in other areas.

Regarding soft deletion we have the system in place, I mentioned the issue because the problem doesn't stem from database constraint issues, rather, it relates more to maintaining data coherence in history, stats, etc..

While we don't currently prioritize this issue, we have plans to address it eventually in order to prevent duplicated questions from being stored under the same subject.
